chapter0-What-is-Java/terminology.md

```diff
@@ -1,19 +1,21 @@
 **Compiler:** The compiler converts source code into a  low-level program.
-In most compiled languages, the file containing the resulting low-level code is called an object  file.
+In most compiled languages, the file containing the resulting low-level code is called an object file.  Compiled languages such as C and C++ use a platform-specific compilers, which output platform-specific object code.
 
-**Linker:** A collection of object files are linked together to create an executable file (i.e., the operating system can load the executable into RAM to run the program). Another term for an executable is "(relocatable) machine code".
+**Platform** A specific target machine that code can be executed on. For example, if you compile a C++ program on a Windows machine, the executable file can be copied to any other machine but it will only run on other Windows machines but never another machine (e.g., a Mac or a Linux machine). A platform is determined by the target machine's hardware and operating system.  Different processor types, for example, will utilise different codes for the same operations.
```

Comment:
A specific target machine that can execute the code.
For example, if you compile a C++ program for a 64-bits Windows machine, the executable file can be copied to any other machine, but it can only run on other similar 64-bits Windows machines. It cannot run on a 32-bits Windows, a Mac or a Linux machine.
The differences are due to different hardware and operating system, that utilise different codes for the same operations.

Note: The machine you compile *on* does not necessarily need to be the machine you compile *for*. For example, in robotics, it is common to compile on a computer before loading the executable on a robot.